shards/eval.go

```diff
@@ -95,7 +97,7 @@ func (s *typeRepoSearcher) eval(ctx context.Context, q query.Q) (query.Q, error)
 		}
 
 		var rl *zoekt.RepoList
-		rl, err = s.Streamer.List(ctx, rq.Child)
```

Comment:
for future, but minimal would be useful here since we are calling list here to find which repos to search.

Reply:
We'd have to make RepoSet work with IDs at that point. Or introduce a RepoIDSet. What's the approach you have in mind?

Reply:
yup. We mainly use RepoBranches so that would also need be needed.